File: BehaviorTreesLib/Composite.cpp
#include "Composite.h"

namespace fluentBehaviorTree
{
	void Composite::addChild(Node & n)
	{
		mChildren.push_back(&n);
	}

	void Composite::open()
	{
		Node::open();
		for each(auto child in mChildren)
		{
			child->setResult(EStatus::RUNNING);
		}
	}
}

File: C#/BehaviorTree/Branch/Composite/Composite.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FluentBehaviorTree
{
    public abstract class Composite : Branch
    {
        protected List<Node> children;

        public override void AddChild(Node n)
        {
            children.Add(n);
        }

        protected override void Open()
        {
            base.Open();

            foreach (var child in children)
            {
                child.Result = Status.RUNNING;
            }
        }
    }
}
